files/default/neutron-ha-tool.py

```diff
@@ -667,7 +667,10 @@ def migrate_router(qclient, router, agent, target,
     LOG.debug("Removed router from agent=%s" % agent['id'])
 
     # ensure it is removed
-    if router in list_routers_on_l3_agent(qclient, agent['id']):
+    router_ids = [
+        r['id'] for r in list_routers_on_l3_agent(qclient, agent['id'])
+    ]
```

Comment:
If we convert it into a generator we can have some bit of speed, because the `in` will not get all the ids (it will stop the `for` in the moment that it finds the router id)

Anyway +1